issuer-web/src/router/guards/pres-req.ts

```diff
@@ -0,0 +1,31 @@
+import IssuerStore from "@/store";
+import { Route } from "vue-router";
+import { decode, verify } from "jsonwebtoken";
+import { AppConfig } from '@/models/appConfig';
+import * as ConfigService from "@/services/config";
+
+
+
+export default async function hasPresReq(
+  to: Route,
+  from: Route,
+  next: Function
+) {
+  const config: any = await ConfigService.getAppConfig();
+  IssuerStore.getInstance()
+    .dispatch("oidcStore/getOidcUser")
+    .then((result: any) => {
+        //grab req id tokens if they exist
+        const idToken = decode(result?.id_token) as any
+        const reqId = idToken?.pres_req_conf_id
+        const confReqId = config.authentication.oidcSettings?.extraQueryParams?.pres_req_conf_id
+
+        //unauthorized if pre_req_conf_id are both set and mismatch
+        if(reqId && confReqId && (reqId !== confReqId)){
```

Comment:
This seems wrong. Checking for `reqId` to be truthy will cause the guard to not work as expected (send the user to the `unauthorized` page) when the `pres_req_conf_id` query parameter is configured in the `oidcSettings` but nothing is returned in the token after authentication.

The guard should prevent access if `confReqId` is set and it's value does not match the value of `pres_req_conf_id` in the token.